test_chameleon.py

```diff
@@ -11,43 +14,73 @@ def test_basic_json_service(self):
         """
         Mock server with a single URL returning a successful JSON response
         responds to queries as expected.
-        Port defaults to 8080.
         """
-        settings_list = [{"url": "/path1", 
-                          "status_code": 200, 
-                          "content_type": "application/json", 
-                          "content": '{"foo": "bar"}'}]
-        chameleon.start(settings_list)
-        response = requests.get('http://localhost:8080/path1')
+        settings_list = [{
+            "url": "/path1",
+            "status_code": 200,
+            "content_type": "application/json",
+            "content": '{"foo": "bar"}'
+        }]
+
+        chameleon.start(settings_list, port=self.port)
+        response = requests.get('http://localhost:{}/path1'.format(self.port))
+
+        # basic content stuff
         self.assertEqual(response.status_code, 200)
-        self.assertEqual(response.content, '{"foo": "bar"}')
-        self.assertEqual(response.json()['foo'], 'bar')  # ensure response decoded as JSON.
+        self.assertEqual(response.content, b'{"foo": "bar"}')
+
+        # ensure response decoded as JSON.
+        self.assertEqual(response.json()['foo'], 'bar')
         self.assertEqual(response.headers['content-type'], 'application/json')
 
+    def test_content_type_xml(self):
+        settings_list = [{
+            "url": "/path1",
+            "status_code": 200,
+            "content_type": "application/xml",
+            "content": '<foo>bar</foo>'
+        }]
+        chameleon.start(settings_list, port=self.port)
+        response = requests.get('http://localhost:{}/path1'.format(self.port))
+        self.assertEqual(response.content, b'"<foo>bar</foo>"')
```

Comment:
Since this is the only XML test, can we add an assert for content-type too?
`self.assertEqual(response.headers['content-type'], 'text/xml')`